### PR TITLE
Refactor for visitor updater

### PR DESCRIPTION
**Pull request by paniagua**
It included:

Wrapping the updater into the sm object to avoid
clashes with other functions within the website

Cleaner (not perfect) way of get and post visitor
information

Pull the visitor lateset information before updating
the visitor information with new values. This avoids
loosing date from other sources (e.g. operator console)

@justindipietro @batshoes

### CODE REVIEW

**Comment by paniagua** [comment]
This refactor is far from perfect. 

Unfortunately I did not find any test (tested manually)
I think some tests should be added...

File: tracker/sm-website-update-visitor.js [hunk 1]
```diff
@@ -1,147 +1,153 @@
...
-        } else {
+        if ($(this).val().length > 0 ) {
           name = $('#First-Name-3').val() + " " + $('#Last-Name-3').val()
-          updateVisitorDetails("name", name);
+          updateVisitorInformation("name", name);
         }
       });
 
       $('#First-Name-2, #Last-Name-2').change(function(){
-        if ($(this).val() === "") {
```
Comment (paniagua) [inline tracker/sm-website-update-visitor.js:48]: this kind of conditions do not make sense - a simple negation of the condition does

File: tracker/sm-website-update-visitor.js [hunk 2]
```diff
@@ -1,147 +1,153 @@
...
+        if ($(this).val().length > 0) {
           phone = $(this).val();
-          updateVisitorDetails("phone", phone);
+          updateVisitorInformation("phone", phone);
         };
       });
     }
-  ); 
-  
-function getHeaders(){
```
Comment (paniagua) [inline tracker/sm-website-update-visitor.js:96]: this function sets some global values. it does not make sense then to be function.

File: tracker/sm-website-update-visitor.js [hunk 3]
```diff
@@ -1,147 +1,153 @@
...
+      }
+    );
+  }
+
+
+  getCurrentVisitor = function() {
+
+    return new Promise(function(resolve, reject) {
+     var xhr = new XMLHttpRequest();
+     var API_DOMAIN = "api.salemove.com"
```
Comment (batshoes) [inline tracker/sm-website-update-visitor.js:56]: What is the point of extracting the different parts of the url and combining them when for this particular script it is all defined under the visitor endpoint ?

File: tracker/sm-website-update-visitor.js (hunk 1 above)
Comment (batshoes) [inline tracker/sm-website-update-visitor.js:48]: You mean the nature of 
if true
  do nothing
else 
  do the code
and if so, is that the only difference between > 0 and "" in this case?

File: tracker/sm-website-update-visitor.js [hunk 4]
```diff
@@ -1,147 +1,153 @@
...
+          },
+        url: "https://api.salemove.com/visitor",
+        data: visitor
+        });
+      }, function(error) {
+        console.log('An error occurred: ', error);
+      }
+    );
+  }
+
```
Comment (batshoes) [inline tracker/sm-website-update-visitor.js:50]: How does the above function negate the overwriting of previously stored information? 
If name: "james" 
and then this function is run with an alternative name, would it not still update it?

File: tracker/sm-website-update-visitor.js (hunk 1 above)
Comment (paniagua) [inline tracker/sm-website-update-visitor.js:48]: the idea is to check what you are interested on. You are interested in knowing when there is a new value that is not empty string. checking `> 0` does both.

File: tracker/sm-website-update-visitor.js (hunk 3 above)
Comment (paniagua) [inline tracker/sm-website-update-visitor.js:56]: I tested it in my local environment, it was clearer what part i needed to change. In a later improvement they can be extracted into a configuration object and they can be injected.

File: tracker/sm-website-update-visitor.js (hunk 4 above)
Comment (paniagua) [inline tracker/sm-website-update-visitor.js:50]: if the attribute is not present it is not updated.

**Pull request merged**